tests/TestContextTest.cs

```diff
@@ -0,0 +1,186 @@
+﻿using System;
+using System.Threading.Tasks;
+using Egil.RazorComponents.Testing.EventDispatchExtensions;
+using Egil.RazorComponents.Testing.SampleComponents;
+using Egil.RazorComponents.Testing.SampleComponents.Data;
+using Moq;
+using Shouldly;
+using Xunit;
+
+namespace Egil.RazorComponents.Testing
+{
+    public class TestContextTest
```

Comment:
```suggestion
    public class TestContextTest : ComponentTestFixture 
```
By inherting from `ComponentTestFixture`, we get a TextContext implicitly, follow the convention from the other tests that also does this, and do not have to deal with disposing of the context after each test. It is handled by the fixture.

Reply:
Actually I don't like this.

1. I don't think the test framework should be used to test itself. Or at least tests of `TestContext` should not depend on `ComponentTestFixture` that depends on `TestContext`.
2. Inheriting `ComponentTestFixture` from `TestContext` can lead to flaky tests (or tests depending on order). For example in one of my tests I setup Services. And this setup of services is then carried on to other tests. But it depends on order of execution to which tests.

In general test classes should be stateless and new state should be constructed in each test. So `ComponentTestFixture` is anti-pattern.

Reply:
Order of tests does not matter at all since xUnit creates a new instance of the `TestContextTest` for each test/fact it executes, and when the test is done, it runs the dispose method inherited by `TestContextTest`.

That means every test has it's own test context.

As to the first point, I get what you are saying. But since `ComponentTestFixture` just provides helper methods for tests, and it is a test context, I do think it makes sense to do it like this.

Reply:
Oh, yes. I forgot that xUnit works differently than NUnit and MSTest. I will update the pattern then.